client_monitor.go

```diff
@@ -26,32 +26,31 @@ func newClientMonitor() *clientMonitor {
 		// so that the senders can avoid blocking and to account for some delivery delay.
 		snapshotBuffer:  make(chan snapshotAndSubscribers, 100),
 		currentSnapshot: componentstatus.ClientSnapshot{Producers: make(map[string]componentstatus.Status)},
-
-		shutdownOnce: &sync.Once{},
-		shutdownCh:   make(chan struct{}),
-		doneCh:       make(chan struct{}),
```

Comment:
Lets us get rid of a whole bunch of vars, which is nice. We add a `StartStopStress` test case also, so we're now much more certain that this code works than before.